File: apps/android-client/src/BridgeHealthIndicator.kt
package com.gogate.android

data class BridgeHealthViewModel(
    val healthy: Boolean,
    val reason: String,
    val badgeColor: String,
    val label: String
)

object BridgeHealthIndicator {
    fun from(healthy: Boolean, reason: String): BridgeHealthViewModel {
        return if (healthy) {
            BridgeHealthViewModel(true, reason, "#2ECC71", "Bridge Healthy")
        } else {
            val lowered = reason.lowercase()
            val color = if (lowered.contains("timeout") || lowered.contains("poll")) "#F39C12" else "#E74C3C"
            BridgeHealthViewModel(false, reason, color, "Bridge Degraded")
        }
    }
}

File: apps/windows-client/src/BridgeHealthIndicator.cs
namespace GoGate.WindowsClient;

public record BridgeHealthViewModel(bool Healthy, string Reason, string BadgeColor, string Label);

public static class BridgeHealthIndicator
{
    public static BridgeHealthViewModel From(bool healthy, string reason)
    {
        if (healthy)
            return new BridgeHealthViewModel(true, reason, "#2ECC71", "Bridge Healthy");

        var lowered = (reason ?? string.Empty).ToLowerInvariant();
        var color = lowered.Contains("timeout") || lowered.Contains("poll") ? "#F39C12" : "#E74C3C";
        return new BridgeHealthViewModel(false, reason, color, "Bridge Degraded");
    }
}
